bfq.io.weight followups and cleanup

Werkov opened the pull request:
One main changes in this series:
- <del>support IODeviceWeight for the BFQ cgroup attribute too</del> -> #21795,
- add debug message about scaled BFQ weight.

The rest is just tidying up the surrounding code.

yuwata reviewed (commented):
LGTM except for the last commit. 

@poettering, if I understand and remember your previous comments somewhere, you do not like that pid1 touches bfq.weight. WDYT?

Werkov commented:
Rebased on the current `main`, some fine touches on the logged messages, extracting the per-device setting into a separate PR.

yuwata commented on src/core/cgroup.c:
```diff
@@ -1279,28 +1279,36 @@ assert_cc(BFQ_WEIGHT(10000) == 1000);
-        (void) set_attribute_and_warn(u, "io", "io.bfq.weight", buf);
+        bfq_weight = BFQ_WEIGHT(weight);
+        xsprintf(buf, "%" PRIu64 "\n", bfq_weight);
+        if (!set_attribute_and_warn(u, "io", "io.bfq.weight", buf))
```
`set_attribute_and_warn()` returns negative errno on failure. And, we do not allow to implicit cast to boolean.
Please do like the following:
```
if (set_attribute_and_warn >= 0)
        log_unit_debug(u, "...");
```

yuwata commented on src/core/cgroup.c:
```diff
@@ -1279,28 +1279,36 @@ assert_cc(BFQ_WEIGHT(10000) == 1000);
-        (void) set_attribute_and_warn(u, "blkio", "blkio.bfq.weight", buf);
+        bfq_weight = BFQ_WEIGHT(weight);
+        xsprintf(buf, "%" PRIu64 "\n", bfq_weight);
+        if (!set_attribute_and_warn(u, "blkio", "blkio.bfq.weight", buf))
```
Also here.

yuwata commented on src/core/cgroup.c:
```diff
@@ -1047,6 +1047,26 @@ static uint64_t cgroup_weight_io_to_blkio(uint64_t io_weight) {
+
+        xsprintf(buf, "%" PRIu64 "\n", bfq_weight);
+
+        if (!set_attribute_and_warn(u, controller, p, buf))
```
Also here.

Werkov commented:
Force pushed changes:
- rebased in current `main`,
- addressed review feedback,
- don't pollute log if scaling factor 1,
- added `Block` prefix in the log message.

yuwata reviewed (approved):
LGTM.
